git_machete/client.py

```diff
@@ -1682,21 +1682,23 @@ def checkout_github_prs(self, pr_no: int) -> None:
         remote: str
         remote, (org, repo) = self.__derive_remote_and_github_org_and_repo()
         debug('checkout_github_pr()', f'organization is {org}, repository is {repo}')
+        print(f"Fetching {remote}...")
+        self.__git.fetch_remote(remote)
+        self.flush_caches()
 
-        all_prs: List[GitHubPullRequest] = derive_pull_requests(org, repo)
-        pr = utils.find_or_none(lambda pr: pr.number == pr_no, all_prs)
-
+        pr = get_pull_request_by_number_or_none(pr_no, org, repo)
         if not pr:
             raise MacheteException(f"PR #{pr_no} is not found in repository `{org}/{repo}`")
+        if '/'.join([remote, pr.head]) not in self.__git.get_remote_branches():
+            raise MacheteException(f"Could not check out PR #{pr_no} because its head branch `{pr.head}` is already deleted from `{remote}`.")
```

Comment:
TBD: Here we could `fetch pull/$ID/head` (also for #242) and return here instead of rising an exception. This would also allow to checkout pull requests from forks, but it might be difficult to find base branch (for example: very old PR #73 can't point to any branch at the moment cause its outdated, but #241 could point to develop)

Reply:
Good spot with this corner case (#73)... I'll address on #242 🤔